room.go

```diff
@@ -152,6 +163,7 @@ func NewRoom(callback *RoomCallback) *Room {
 		sidDefers:          make(map[livekit.ParticipantID][]func(*RemoteParticipant)),
 		callback:           NewRoomCallback(),
 		sidReady:           make(chan struct{}),
+		connectionState:    ConnectionStateDisconnected,
```

Comment:
Maybe a 'ConnectionStateInit' as disconnected is terminal and it would be good to have one meaning for disconnected value.

Reply:
it's not a bad idea.. though if we were to add the state we should do so across clients. Currently the other client SDKs do not expose this `initial` state: https://github.com/livekit/client-sdk-js/blob/main/src/room/Room.ts#L104

Reply:
so I'll merge this one without. we can add in the future if we decide to change.